synapseclient/models/table.py

```diff
@@ -1,30 +1,1088 @@
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            The Table instance stored in synapse.
+
+        Example: Getting metadata about a table using id
+            Get a table by ID and print out the columns and activity. `include_columns`
+            defaults to True and `include_activity` defaults to False. When you need to
+            update existing columns or activity these need to be set to True during the
+            `get` call, then you'll make the changes, and finally call the
+            `.store()` method.
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Table
+
+            syn = Synapse()
+            syn.login()
+
+            table = Table(id="syn4567").get(include_activity=True)
+            print(table)
+
+            # Columns are retrieved by default
+            print(table.columns)
+            print(table.activity)
+            ```
+
+        Example: Getting metadata about a table using name and parent_id
+            Get a table by name/parent_id and print out the columns and activity.
+            `include_columns` defaults to True and `include_activity` defaults to
+            False. When you need to update existing columns or activity these need to
+            be set to True during the `get` call, then you'll make the changes,
+            and finally call the `.store()` method.
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Table
+
+            syn = Synapse()
+            syn.login()
+
+            table = Table(name="my_table", parent_id="syn1234").get(include_columns=True, include_activity=True)
+            print(table)
+            print(table.columns)
+            print(table.activity)
+            ```
+        """
+        return self
+
+    def delete(self, *, synapse_client: Optional[Synapse] = None) -> None:
+        """Delete the entity from synapse. This is not version specific. If you'd like
+        to delete a specific version of the entity you must use the
+        [synapseclient.api.delete_entity][] function directly.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            None
+
+        Example: Deleting a table
+            Deleting a table is only supported by the ID of the table.
+
+            ```python
+            from synapseclient import Synapse
+
+            syn = Synapse()
+            syn.login()
+
+            Table(id="syn4567").delete()
+            ```
+        """
+        return None
+
+    @staticmethod
+    def query(
+        query: str,
```

Comment:
We may remove this https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1175/files#r1999374719